accounts/views.py

```diff
@@ -0,0 +1,141 @@
+    the session is logged and the user is redirected to dashboard
+    else user is redirected to loginpage with error
+    """
+
+    def post(self, request, *args, **kwargs):
+        if 'email' in request.session:
+            del request.session['email']
+        email = request.POST['email']
+        password = request.POST['password']
+        user = authenticate(request, email=email, password=password)
+        if user is not None and user.is_active:
+            login(request, user)
+            if not request.POST.get('remember-me', None):
+                # logic for making the user logout as soon as browser is closed
+                # if remember me is not checked on login
+                request.session.set_expiry(0)
+                request.session.modified = True
+            messages.success(request, 'Login Successful!')
+        else:
+            request.session['email'] = email
+            messages.error(
+                request, 'Incorect Email/Password! Please try again!')
+            return redirect("user_profile:index")
+        return redirect("user_profile:dashboard")
+
+
+class Logout(View, LoginRequiredMixin):
+    """ VIew for Handling logout
+    Only supports Post requests
+    logouts the user when endpoint is accessed else
+    throws error. User must be logged in to logout
+    """
+    login_url = 'user_profile:index'
+
+    def post(self, request, *args, **kwargs):
+        try:
+            logout(request)
+            messages.success(request, 'Logout Successful!')
+        except Exception as e:
+            messages.error(request, "Something went Wrong! {}".format(e))
+        return redirect('user_profile:index')
+
+
+class Register(View):
+    """View for User registration
+    only supports POST Method
+    creates the necessary db objects
+    logs the user in and redirect the user
+    to the dashboard else redirects back to
+    signup page
+    """
+
+    def post(self, request, *args, **kwargs):
+        registered_feed_content = '{full_name} has joined UBook'
+        if 'email' in request.session:
+            del request.session['email']
+
+        email = request.POST['email']
+        password1 = request.POST['password1']
+        password2 = request.POST['password2']
+        first_name = request.POST['firstname']
+        last_name = request.POST['lastname']
+        birthday = request.POST['birthday']
+        gender = request.POST['gender']
+        if password1 != password2:
+            messages.warning(
+                request, "Password and Confirm Password do not Match!")
+            return redirect('user_profile:index')
+        email = managers.CustomUserManager.normalize_email(email)
+        user = None
+        try:
+            user = models.CustomUser.objects.create(
+                email=email, first_name=first_name, last_name=last_name, is_active=True)
+            user.set_password(password1)
+            user_profile = UserProfile.objects.create(
```

Comment:
lets use signals here as well please.

Reply:
fixed

Reply:
fixed added signals for both user registeration post and user profile object creation

Reply:
Lets not create a custom signal here since we can use the oist_save signal to handle the special case as well.